connect/src/spirc.rs

```diff
@@ -724,6 +730,13 @@ impl SpircTask {
         }
         cs.send();
     }
+
+    fn cache_volume(&self, volume: u16) {
```

Comment:
Change this to `set_volume` and add to this function:
```rust
self.device.set_volume(volume as u32);
self.mixer.set_volume(volume_to_mixer(volume, self.linear_volume));
```

This way the calls can be removed from the other places, since we always need to update all three values